Eval/Tokenizer.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace Eval;
+
+#region Class tokenizer ---------------------------------------------------------------------------
+/// <summary>Splits the expression into individual tokens</summary>
+class Tokenizer {
+   public Tokenizer (Evaluator eval, string text) {
+      mText = text; mN = 0; mEval = eval;
+   }
+   readonly Evaluator mEval;  // The evaluator that owns this 
+   readonly string mText;     // The input text we're parsing through
+   int mN;                    // Position within the text
+
+   #region Methods --------------------------------------------------
+   /// <summary>Gets the token one by one from the expression</summary>
+   /// <returns>Individual token</returns>
+   public Token Next () {
+      while (mN < mText.Length) {
+         char ch = char.ToLower (mText[mN++]);
+         switch (ch) {
+            case ' ' or '\t': continue;
+            case (>= '0' and <= '9') or '.': return GetNumber ();
+            case '(' or ')':
+               mEval.BasePriority += ch == '(' ? 10 : -10;
+               return new TPunctuation (ch);
+            case '+' or '-' or '*' or '/' or '^' or '=':
+               if (ch is '-' or '+' && (mN - 1 == 0 || mText[mN - 2] is '+' or '-' or '*' or '/' or '^' or '(' || mFuncs.Contains (mText[0..(mN - 1)]))) return new TOpUnary (mEval, ch);

```

Comment:
```suggestion
               if (ch is '-' or '+' && (mN - 1 == 0 || mText[mN - 2] is '+' or '-' or '*' or '/' or '^' or '(' || mFuncs.Contains (mText[0..(mN - 1)]))) return new TOpUnary (mEval, ch);

```
Why is this check for all arithmetic operators done here? Should check only for - or + right?